File: src/test/java/data/NameGeneratorTest.java
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotEquals;
import org.junit.Test;
import org.junit.Ignore;

import fortress.data.NameGenerator;
import fortress.data.SubIntNameGenerator;
import java.util.Set;

public class NameGeneratorTest {
    @Test
    public void SubIntTest() {
        Set<String> forbid = Set.of("sk_1");
        SubIntNameGenerator gen = new SubIntNameGenerator(forbid, 0);
        assertEquals("sk_0", gen.freshName("sk"));
        assertEquals("sk_2", gen.freshName("sk"));
        assertEquals("sk_3", gen.freshName("sk"));
    }
}

File: src/test/scala/data/NameGeneratorTest.scala
import org.scalatest._
import org.junit.runner.RunWith
import org.scalatest.junit.JUnitRunner

import fortress.data.IntSuffixNameGenerator

@RunWith(classOf[JUnitRunner])
class NameGeneratorTest extends FunSuite with Matchers {
    
    test("integer suffix name generator") {
        val forbid = Set("sk_3")
        val gen = new IntSuffixNameGenerator(forbid, 2)
        
        gen.freshName("sk") should be ("sk_2")
        gen.freshName("sk") should be ("sk_4")
        gen.freshName("sk") should be ("sk_5")
    }
}
